api/src/main/groovy/ncei/onestop/api/service/MetadataParser.groovy

```diff
@@ -261,47 +261,43 @@ class MetadataParser {
   }
 
   static Map parseSpatialInfo(GPathResult metadata) {
-    def spatialBounding = [:]
-    def isGlobal = null
     def space = metadata.identificationInfo.MD_DataIdentification.extent.EX_Extent.'**'.find { e ->
       e.@id.text() == 'boundingExtent'
     }.geographicElement
     def bbox = space.'**'.find { it -> it.name() == 'EX_GeographicBoundingBox' }
-    if (bbox) {
 
-      spatialBounding.put('type', 'envelope')
-      spatialBounding.put('coordinates', parseCoords(bbox))
-              //)
-      isGlobal = checkIsGlobal(bbox)
-    }
-    return ["spatialBounding":spatialBounding, "isGlobal" : isGlobal]
+    def spatialBounding = parseBounding(bbox)
+    def isGlobal = checkIsGlobal(spatialBounding)
+
+    return ["spatialBounding": spatialBounding, "isGlobal": isGlobal]
   }
 
-  static def parseCoords(def bbox){
-    def westBound = (bbox.westBoundLongitude == "null" ||  bbox.westBoundLongitude == "") ? null : bbox.westBoundLongitude.Decimal.toFloat()
-    def eastBound = (bbox.eastBoundLongitude == "null" ||  bbox.eastBoundLongitude == "") ? null : bbox.eastBoundLongitude.Decimal.toFloat()
-    def northBound = (bbox.northBoundLatitude == "null" || bbox.northBoundLatitude == "")  ? null : bbox.northBoundLatitude.Decimal.toFloat()
-    def southBound = (bbox.southBoundLatitude == "null" || bbox.southBoundLatitude == "") ? null : bbox.southBoundLatitude.Decimal.toFloat()
+  static def parseBounding(def bbox) {
+    if (!bbox) { return null }
```

Comment:
Before, no bbox meant spacialBounding defaulted to [:], not null. Is that going to be an issue?

Reply:
Tested against the ES API directly and sending null for the spatialBounding field works just fine, and actually sending an empty object doesn't work. I think the Java Client is making it work for us either way, though this way will port more easily to the ES 5.x branch.